common/djangoapps/xblock_django/user_service.py

```diff
@@ -69,6 +73,15 @@ def _convert_django_user_to_xblock_user(self, django_user):
             xblock_user.opt_attrs[ATTR_KEY_USER_ID] = django_user.id
             xblock_user.opt_attrs[ATTR_KEY_USERNAME] = django_user.username
             xblock_user.opt_attrs[ATTR_KEY_USER_IS_STAFF] = django_user.user_is_staff
+            try:
+                user_preferences = get_user_preferences(django_user)
+                xblock_user.opt_attrs[ATTR_KEY_USER_PREFERENCES] = {
+                    pref: user_preferences.get(pref)
+                    for pref in USER_PREFERENCES_WHITE_LIST
+                    if pref in user_preferences
+                }
+            except UserNotFound:
```

Comment:
Is this an expected error case?  Is there a reason we need to handle this explicitly?